mozilla-ai/lumigator
# Local mzai-platform development with devcontainers, vscode, and compose

Building on the work @binaryaaron and I were doing last week, this setup works by building all docker-compose locally and then editing via devcontainers. This setup assumes development on M2 Mac. 

Note that the web server does hot reloading for quick local development - we may want a separate production Dockerfile that disables this. 

To test:

1.  pull branch and go to root
2. `make local-arm`
3. Open the codebase in vscode, you should get a prompt like this, then reopen in container and open your web browser to http://127.0.0.1/docs to test changes.  
<img width="717" alt="Screenshot 2024-06-17 at 1 23 02 PM" src="https://github.com/mozilla-ai/mzai-platform/assets/3837836/1b5ab323-630f-40d3-9055-f1774b7fedbb">
